internal/services/machinelearning/machine_learning_compute_cluster_resource.go

```diff
@@ -206,32 +206,49 @@ func resourceComputeClusterCreate(d *pluginsdk.ResourceData, meta interface{}) e
 		computeClusterAmlComputeProperties.Subnet = &machinelearningcomputes.ResourceId{Id: subnetId.(string)}
 	}
 
+	// NOTE: The 'AmlCompute' 'ComputeLocation' field should always point
+	// to configuration files 'location' field...
```

Comment:
we should add more detail here about why this is and link to a rest API specs issue detailing what cluster.location and cluser.compute.location actually mean, that they are backwards? and the api is both confusing and is not consistent with the rest of azure